packages/backend-storage/src/factory.ts

```diff
@@ -46,7 +55,143 @@ class AmplifyStorageFactory
         getInstanceProps
       );
     }
-    return constructContainer.getOrCompute(this.generator) as AmplifyStorage;
+    const amplifyStorage = constructContainer.getOrCompute(
+      this.generator
+    ) as AmplifyStorage;
+
+    /*
+     * only call Aspects once,
+     * otherwise there will be the an error -
+     * "there is already a construct with name 'storageRegion'"
+     */
+    const aspects = Aspects.of(Stack.of(amplifyStorage));
+    if (!aspects.all.length) {
+      aspects.add(
+        new StorageOutputsAspect(getInstanceProps.outputStorageStrategy)
+      );
+    }
+
+    return amplifyStorage;
+  };
+}
+
+/**
+ * StorageValidator class implements the IAspect interface.
+ */
+export class StorageOutputsAspect implements IAspect {
+  isStorageProcessed = false;
+  node: IConstruct;
+  outputStorageStrategy;
+  /**
+   * Constructs a new instance of the StorageValidator class.
+   */
+  constructor(
+    outputStorageStrategy: BackendOutputStorageStrategy<StorageOutput>
+  ) {
+    this.outputStorageStrategy = outputStorageStrategy;
+  }
+
+  /**
+   * Visit the given node.
+   * If the node is an AmplifyStorage construct, we will traverse its siblings in the same stack
+   * @param node The node to visit.
+   */
+  public visit(node: IConstruct): void {
+    if (!(node instanceof AmplifyStorage)) {
+      return;
+    }
+    const storageInstances = Stack.of(node).node.children.filter(
+      (el) => el instanceof AmplifyStorage
+    );
+    const storageCount = storageInstances.length;
+
+    /**
+     * only store traverse the siblings once to store the outputs,
+     * otherwise there will be the an error -
+     * "there is already a construct with name 'storageRegion'"
+     */
+    if (this.isStorageProcessed) {
+      return;
+    }
```

Comment:
move this in the beginning for early return

Reply:
Done!